lib/palindrome_check.rb

```diff
@@ -1,7 +1,29 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) with n being the length of the input string
+# Space complexity: O(1)
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  if my_phrase == nil
+    return false
+  elsif my_phrase == ""
+    return true
+  end
+
+  i = 0
+  k = 1
+  while i < my_phrase.length
+    
+    if my_phrase[i] == ' ' || my_phrase[-k] == ' '
```

Comment:
What if only the left side has a few leading space?

Instead I would suggest making 2 while loops

```ruby
while my_phrase[i] == ' '
  i += 1
end
```